core/shared/src/main/scala/laika/parse/code/languages/HaskellSyntax.scala

```diff
@@ -0,0 +1,43 @@
+package laika.parse.code.languages
+
+import cats.data.NonEmptyList
+import laika.bundle.SyntaxHighlighter
+import laika.parse.code.common.Identifier.IdParser
+import laika.parse.code.common.{Comment, Keywords, NumberLiteral, StringLiteral}
+import laika.parse.code.languages.ScalaSyntax.charEscapes
+import laika.parse.code.{CodeCategory, CodeSpanParser}
+import laika.parse.text.CharGroup.{digit, lowerAlpha, upperAlpha}
+
+object HaskellSyntax extends SyntaxHighlighter {
+  /** The names of the language (and its optional aliases) as used in text markup */
+  override def language: NonEmptyList[String] = NonEmptyList.of("hs", "haskell")
+
+  val comment: CodeSpanParser = Comment.singleLine("--") ++ Comment.multiLine("{-", "-}")
+
+  val keywords =
+    Keywords("as", "case", "data", "data family", "data instance", "default", "deriving",
+      "deriving instance", "do", "else", "forall", "foreign", "hiding", "if", "import", "in",
+      "infix", "infixl", "infixr", "instance", "let", "mdo", "module", "newtype", "of",
+      "proc", "qualified", "rec", "then", "type", "type family", "type instance", "where"
+    )
+
+  val stringLiteral =
+    StringLiteral.singleLine('"').embed(charEscapes)
+
```

Comment:
Adding `charLiteral` would be nice. You are right that none of these syntax definitions are supposed to cover 100%, but char literals are supported in all other languages, so it would make it more consistent.